web-animations-2/Overview.bs

```diff
@@ -2353,6 +2354,142 @@ Items sorted earlier are executed before those sorted later.
+
+Issue: Should there be any effect of triggers on scroll-driven animations?
+
+### Animation Trigger State ### {#trigger-state}
+
+An [=animation trigger=] |trigger| with [=animation trigger type=] |type|
+and an associated [=animation=] |animation|
+has an internal <dfn export lt="animation trigger state">state</dfn> that can be one of the following:
+
+<dl dfn-for="animation trigger state" dfn-type=value>
+  <dt><dfn>idle</dfn>
+  <dd>
+    The [=animation effect=] associated |animation| remains in
+    its [=animation effect/before phase=] and stays at zero [=animation/start time=].
+
+  <dt><dfn>primary</dfn>
+  <dd>
+    The ''primary'' effect defined by |type| is applied to |animation|.
+
+  <dt><dfn>inverse</dfn>
+  <dd>
+    The ''inverse'' effect defined by |type| is applied to |animation|.
+
+</dl>
+
+Issue: Do we need a formal resolution on the spec of the idle state?
+
+The [=animation trigger state=] |state| of an [=animation trigger=] |trigger|
+is controlled by the [=animation attachment range|attachment range=] it is attached to,
+given a flag |did trigger|, as follows:
+
+<dl class=switch>
+  : If |trigger|’s [=local time=] is [=unresolved=],
+  ::
+    Then |state| is ''animation trigger state/idle'', and |did trigger| is set to false.
+
+  : Otherwise,
+  ::
+
+    <dl class=switch>
+      : If |trigger| is inside its [=active interval=],
+      ::
+        Then |state| is ''animation trigger state/primary'', and |did trigger| is set to true.
+
+      : Otherwise,
+      ::
+
+        <dl class=switch>
+          : If |did trigger| flag is false,
+          ::
+            Then |state| is ''animation trigger state/idle''.
+
+          : Otherwise,
+          ::
+            |state| is ''animation trigger state/inverse''.
+
+        </dl>
+
+    </dl>
+
+</dl>
+
+### Animation Trigger Types ### {#trigger-types}
+
+The behavior an [=animation trigger=] applies to its associated [=animation=]’s
+playback is defined by its <dfn export>animation trigger type</dfn>,
+which can be one of the following:
+
+<dl dfn-type=value dfn-for='animation trigger type'>
+  <dt><dfn>once</dfn>
+  <dd>
+    The ''primary'' behavior is [=play an animation|triggering=] the associated animation.
+
+  <dt><dfn>repeat</dfn>
+  <dd>
+    The ''primary'' behavior is [=play an animation|triggering=] the associated animation.
+    The ''inverse'' behavior is resetting the [=animation trigger state=] back to ''animation trigger state/idle''.
+
+  <dt><dfn>alternate</dfn>
+  <dd>
+    <dl class=switch>
+      : if the |did trigger| flag is false,
+      ::
+        The ''primary'' behavior is [=play an animation|triggering=] the associated animation.
+
+      : Otherwise,
+      ::
+        The ''primary'' behavior is [=reverse an animation|reversing=] the associated animation.
+
+    </dl>
+
+    The ''inverse'' behavior is [=reverse an animation|reversing=] the associated animation.
+
+  <dt><dfn>state</dfn>
```

Comment:
I wonder if something like play-pause would be clearer.

Reply:
That's another option. The idea behind `state` was a reference to `play-state`.

Reply:
Right, I could see play-state as an alternative. State feels too strange, I didn't have any idea what this meant until I read the description :-).